.github/workflows/win.yml

```diff
@@ -83,16 +71,26 @@ jobs:
           echo "set(VCPKG_BUILD_TYPE release)" >> "$TRIPLET_FILE"
           echo "set(VCPKG_DISABLE_COMPILER_TRACKING $VCPKG_DISABLE_COMPILER_TRACKING)" >> "$TRIPLET_FILE"
           cmd.exe /c bootstrap-vcpkg.bat
-          vcpkg.exe install \
-            --vcpkg-root="$VCPKG_ROOT" \
-            --overlay-triplets="$VCPKG_OVERLAY_TRIPLETS" \
-            --triplet="$VCPKG_DEFAULT_TRIPLET" \
-            $VCPKG_INSTALL_OPTIONS
+
+      # make sure we save vcpkg packages even if build fails
+      # note, we don't use vcpkg "command line" mode, but "cmake manifest" mode
+      - name: Restore vcpkg packages
+        id: vcpkg-restore
+        uses: actions/cache/restore@v3
+        with:
+          key: vcpkg=${{ env.VCPKG_VERSION }}-msvc=${{ env.MSVC_VERSION }}-json=${{ hashFiles('vcpkg.json') }}
+          path: |
+            vcpkg/*
+            !vcpkg/downloads
+            !vcpkg/docs
+            !vcpkg/buildtrees
+            vcpkg/downloads/tools
```

Comment:
pff, this was super super annoying to get right.. they do have some serious growing pains, think I got involved too early and things change around while they can't break stuff that previously worked.. or they have the "valhalla syndrome" of trying to cover every use case :wink: but worse docs... 

for the interested reader (and my own future reference), assuming `vcpkg` was installed to the root of the repo:

there's 3 (!!) way how one can run `vcpkg`:
1. classic mode: 
  - **no** `vcpkg.json` in the root of the repo
  - `vcpkg install <pkg1> <pkg2>...` to install packages
  - will install stuff to `./vcpkg/installed`
2. manifest mode (command line):
  - `vcpkg.json` in the root of the repo
  - `vcpkg install` (note, no packages given)
  - will install stuff to `./vcpkg_installed`
3. manifest mode (CMake integrated)
  - `vcpkg.json` in the root of the repo
  - simply `cmake -B build .` will install the packages
  - will install stuff to `build/vcpkg_installed`

what I've been doing was combining 2. & 3.: manually run `vcpkg install` to `./vcpkg_installed` which looks to work just fine and then it installs _another_ time with `cmake -B build .` to `build/vcpkg_installed` which is super quick and seems like it was "cached" from the `vcpkg install`. where it really cached from is `./vcpkg/installed` which was already initiated by `vcpkg install`. but it's not doing anything if you cache the `./vcpkg_installed` directory, it will just be ignored by cmake.

TLDR: the modern cmake way of dealing with vcpkg is to not install anything manually and let cmake configuring take care of it. in terms of caching paths, we want the paths of the vcpkg installation. that finally worked. just we want to ignore the temp build stuff (`vcpkg/downloads` etc), the only interesting thing is located in `vcpkg/installed` (and some cached tool stuff like cmake, pkgconf, 7z.. in `vcpkg/downloads/tools`).

jeez, that is so confusing.. I can hear kevin yell "told ya" hahaha I'm still standing by vcpkg, even there's some questionable design involved..